doc/content/getting-started/events/_index.md

```diff
@@ -59,170 +59,592 @@ See the events of a typical join flow in the example below:
+          },
+          "dev_eui": "70B3D57ED0050EAD",
+          "join_eui": "0000000000000000",
+          "dev_addr": "00FF52E3"
+        }
+      }
+    ],
+    "correlation_ids": [
+      "rpc:/ttn.lorawan.v3.NsJs/HandleJoin:01G76SF01KFGDBGNQKRWG64DBP"
+    ]
+  },
+  {
+    "name": "ns.up.join.cluster.success", //The Network Server indicates that the Join Request to the cluster-local Join Server succeeded
+    "time": "2022-07-05T09:01:06.999074200Z",
+    "identifiers": [
+      {
+        "device_ids": {
+          "device_id": "dev2",
+          "application_ids": {
+            "application_id": "app1"
+          },
+          "dev_eui": "70B3D57ED0050EAD",
+          "join_eui": "0000000000000000",
+          "dev_addr": "004C2AA4"
+        }
+      }
+    ],
+    "data": {
+      "@type": "type.googleapis.com/ttn.lorawan.v3.JoinResponse",
+      "session_keys": {
+        "session_key_id": "AYHNl4A0v5Eu3uRRH34tEA=="
+      }
+    },
+    "correlation_ids": [
+      "gs:conn:01G76RS17HGZZXV740X0N2XDV8",
+      "gs:up:host:01G76RS1T9JM1RGFWT6A0ZEGNB",
+      "gs:uplink:01G76SF01BNW5SKRAND31Z0MGB",
+      "ns:uplink:01G76SF01C64KZ26CKK4SK5G10",
+      "rpc:/ttn.lorawan.v3.GsNs/HandleUplink:01G76SF01BBAMZAQRXYK97RZ8K"
+    ]
+  },
+  {
+    "name": "ns.up.join.process", //The Network Server indicates that the Join Request was successfully processed
+    "time": "2022-07-05T09:01:07.192284684Z",
+    "identifiers": [
+      {
+        "device_ids": {
+          "device_id": "dev2",
+          "application_ids": {
+            "application_id": "app1"
+          },
+          "dev_eui": "70B3D57ED0050EAD",
+          "join_eui": "0000000000000000",
+          "dev_addr": "004C2AA4"
+        }
+      }
+    ],
+    "data": {
+      "@type": "type.googleapis.com/ttn.lorawan.v3.UplinkMessage",
+      "payload": {
+        "join_request_payload": {
+          "join_eui": "0000000000000000",
+          "dev_eui": "70B3D57ED0050EAD",
+          "dev_nonce": "AF7C"
+        }
+      },
+      "rx_metadata": [
+        {
+          "gateway_ids": {
+            "gateway_id": "gw2",
+            "eui": "B827EBFFFE8DB887"
+          }
+        }
+      ],
+      "correlation_ids": [
+        "gs:conn:01G76RS17HGZZXV740X0N2XDV8",
+        "gs:up:host:01G76RS1T9JM1RGFWT6A0ZEGNB",
+        "gs:uplink:01G76SF01BNW5SKRAND31Z0MGB",
+        "ns:uplink:01G76SF01C64KZ26CKK4SK5G10",
+        "rpc:/ttn.lorawan.v3.GsNs/HandleUplink:01G76SF01BBAMZAQRXYK97RZ8K"
+      ]
+    },
+    "correlation_ids": [
+      "gs:conn:01G76RS17HGZZXV740X0N2XDV8",
+      "gs:up:host:01G76RS1T9JM1RGFWT6A0ZEGNB",
+      "gs:uplink:01G76SF01BNW5SKRAND31Z0MGB",
+      "ns:uplink:01G76SF01C64KZ26CKK4SK5G10",
+      "rpc:/ttn.lorawan.v3.GsNs/HandleUplink:01G76SF01BBAMZAQRXYK97RZ8K"
+    ]
+  },
+  {
+    "name": "gs.up.forward", //The Gateway Server receives an information about a successfully processed Join Request from the Network Server and forwards it (where?)
```

Comment:
The GS always forwards to the NS, so this indicates that the uplink was successfully forwarded to the NS. I'm not entirely sure at what point the NS returns "success". I suspect it only does that after completely processing the message, which explains why this event only happens after the NS and JS are done.